Bottom: b7dfa8c66e1dc86225c445ccf919d88ab1bb492c
Top:    cc155d40ac28117d20959cbd97dbc2bbb3c4ffe6
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-06-28 12:40:22 -0700

Refresh of waitall-buggy

---

diff --git a/src/rsocket.c b/src/rsocket.c
index 30727fa..6605df9 100644
--- a/src/rsocket.c
+++ b/src/rsocket.c
@@ -1143,6 +1143,7 @@ ssize_t rrecv(int socket, void *buf, size_t len, int flags)
 
 		ret = 0;
 		if (flags & MSG_PEEK) {
+			// fixme or will peek same data in waitall loop
 			left -= rs_peek(rs, &buf, left);
 			continue;
 		}
@@ -1170,7 +1171,7 @@ ssize_t rrecv(int socket, void *buf, size_t len, int flags)
 			rs->rbuf_offset += rsize;
 			buf += rsize;
 		}
-		rs->rbuf_bytes_avail += len - left;
+		rs->rbuf_bytes_avail += len - left; // <- fixme in waitall loop
 
 	} while ((flags & MSG_WAITALL) && (rs->state & rs_connect_rd) && left);
